packages/image_picker/image_picker_windows/test/image_picker_windows_test.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
+import 'package:file_selector_windows/file_selector_windows.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:image_picker_platform_interface/image_picker_platform_interface.dart';
+import 'package:image_picker_windows/image_picker_windows.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  group('$ImagePickerWindows()', () {
+    final ImagePickerWindows plugin = ImagePickerWindows();
+    final FileSelectorWindows fileSelectorWindows = FileSelectorWindows();
+
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() {
+      fileSelectorWindows.channel
+          .setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        return null;
+      });
+
+      log.clear();
+    });
+
+    test('registered instance', () {
+      ImagePickerWindows.registerWith();
+      expect(ImagePickerPlatform.instance, isA<ImagePickerWindows>());
+    });
+
+    group('images', () {
+      test('pickImage passes the accepted type groups correctly', () async {
+        final XTypeGroup group = XTypeGroup(
+            label: 'images',
+            extensions: <String>[
+              'jpg',
+              'jpeg',
+              'png',
+              'bmp',
+              'webp',
+              'gif',
+              'tif',
+              'tiff',
+              'apng'
+            ]);
+
+        await plugin.pickImage(source: ImageSource.gallery);
+
+        expect(
+          log,
+          <Matcher>[
+            isMethodCall('openFile', arguments: <String, dynamic>{
```

Comment:
You are testing implementation details of `file_selector` here; these tests would all break if FileSelectorWindows changed its implementation (e.g., switched to a Dart implementation, or to using Pigeon).

Even though you can't use `FileSelectorPlatform.instance` in the implementation, you can still write the implementation against a `FileSelectorPlatform` variable, and just manually instantiate it with a `FileSelectorWindows`. Then you can make that instance settable by tests, and insert a mock implementation in these tests and validate against that mock. Then it doesn't matter how `FileSelectorWindows` works, because you'll just be verifying that you're using its API correctly.

Reply:
@stuartmorgan I just pushed the changes. I believe that is what you means, which I agree that is a way better approach.